python/cudf/cudf/core/multiindex.py

```diff
@@ -970,6 +969,16 @@ def __getitem__(self, index):
             result._levels = self._levels
         return result
 
+    @_cudf_nvtx_annotate
+    def _get_level_names(self):
+        """
+        Return a name or list of names with None replaced by the level number.
+        """
+        return [
+            level if name is None else name
+            for level, name in enumerate(self.names)
+        ]
```

Comment:
And this one always returns a list.

Would it be more natural to always return a list, and in the non-multiindex case just have a singleton list that the caller destructures?

Reply:
Yeah IMO if these functions are only needed in `to_frame` and Index vs MultiIndex have their own implementation, I think hardcoding this in the function is sufficient

Reply:
Went ahead and removed these single use functions.